src/components/Buttons/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useState } from 'react';
+import globalStyle from '@/GlobalClasses.module.scss';
+import style from './Carousel.module.scss';
+import ArrowBack from '@/components/Buttons/ArrowButtons/ArrowBack';
+import ArrowNext from '../ArrowButtons/ArrowNext';
+import ArrowButton from '@/components/Buttons/ArrowButtons/ArrowButton';
+import wheatImg from '@/assets/wheat.svg';
+import { initialValues } from '@/services/initialValues';
+import { useFormContext } from 'react-hook-form';
+
+interface CarouselProps {
+  data: string[];
+  registerIndex?: number;
+}
+
+const Carousel = ({ data, registerIndex }: CarouselProps) => {
+  const { register, setValue } = useFormContext();
```

Comment:
**nit:** You are using `useFormContext` inside dumb component which should not be imported here, because it requires FormContext to work. If you would try to reuse in other place outside of form context it it will break. My suggestion is to pass needed values from props